lib/bundler/runtime.rb

```diff
@@ -37,7 +37,17 @@ def setup(*groups)
 
         Bundler.rubygems.mark_loaded(spec)
         load_paths = spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
-        $LOAD_PATH.unshift(*load_paths)
+
+        # See Gem::Specification#add_self_to_load_path (since RubyGems 1.8)
+        insert_index = Gem.load_path_insert_index
```

Comment:
RubyGems' Gem::Specification#add_self_to_load_path adds all load paths for a spec unconditionally.

Bundler's equivalent adds only unique load paths for a spec.

I don't think bundler should have an equivalent of `add_self_to_load_path` until the reason why L39 exists is addressed.  If the bundler version does something different at some point in the future someone will say "oh, these two methods do the same thing, let's have Bundler call the RubyGems one directly" without looking closely and determining they do not do the same thing.

Then something might break.

Keeping these two similar things that behave differently separate prevents such an occurrence.

If L39 can be deleted I support creating the equivalent of `add_self_to_load_path` in Bundler, but it seems orthogonal to this PR's purpose.